src/data/seq/seq.lean

```diff
@@ -45,6 +45,15 @@ def cons (a : α) : seq α → seq α
 /-- Get the nth element of a sequence (if it exists) -/
 def nth : seq α → ℕ → option α := subtype.val
 
+@[simp] lemma nth_cons_zero (s : seq α) (x : α) : nth (cons x s) 0 = some x :=
+by cases s; refl
+
+@[simp] lemma nth_cons_succ (s : seq α) (x : α) (n : ℕ) : nth (cons x s) (n + 1) = nth s n :=
+by cases s; refl
+
+@[simp] lemma val_cons (s : seq α) (x : α) : (cons x s).val = some x :: s.val :=
+by cases s; refl
```

Comment:
It probably makes sense to wait for #15829 which I think makes these true by `rfl`